datasette_dashboards/__init__.py

```diff
@@ -131,9 +131,8 @@ async def dashboard_chart(request, datasette):
     except KeyError:
         raise NotFound(f"Chart does not exist: {chart_slug}")
 
-    db = chart.get("db")
-    if db:
-        database = datasette.get_database(db)
+    database = chart.get("db")
+    if database:
         await check_permission_execute_sql(request, datasette, database)
```

Comment:
As done above with the other call to `check_permission_execute_sql`, I would suggest:

```suggestion
    db = chart.get("db")
    if db:
        database = datasette.get_database(db)
        await check_permission_execute_sql(request, datasette, database.name)
```